python_ta/__init__.py

```diff
@@ -212,66 +210,60 @@ def _get_valid_files_to_check(module_name, local_config):
             yield item
-        elif not os.path.exists(item):
+        elif not os.path.exists(os.path.expanduser(item)):
             try:
                 # For files with dot notation, e.g., `examples.<filename>`
                 filepath = modutils.file_from_modpath(item.split('.'))
                 if os.path.exists(filepath):
                     yield filepath
                 else:
-                    current_reporter.show_file_linted(item)
+                    print(reporter.filename_to_display(item))
                     print('Could not find the file called, `{}`\n'.format(item))
             except ImportError:
-                current_reporter.show_file_linted(item)
-                print('Could not find the file called, `{}`\n'.format(item))
+                print(reporter.filename_to_display(item))
+                print('Could not find the file called 2, `{}`\n'.format(item))
         else:
             yield item  # Check other valid files.
 
 
-def _check(module_name='', level='all', local_config=''):
+def _check(module_name='', level='all', local_config='', output=None):
     """Check a module for problems, printing a report.
     • The `module_name` can take several inputs:
       - string of a directory, or file to check (`.py` extension optional). 
       - list of strings of directories or files -- can have multiple.
       - no argument -- checks the python file containing the function call.
     • `level` is used to specify which checks should be made.
     • `local_config` is a dict of config options or string (config file name).
+    • `output` is an absolute path to capture pyta data output. Default std out.
     """
+
     # Add reporters to an internal pylint data structure, for use with setting
-    # custom pyta options in a Tuple.
+    # custom pyta options in a Tuple, before (re)setting reporter.
     for reporter in REPORTERS:
         VALIDATORS[reporter.__name__] = reporter
+    linter = reset_linter(config=local_config)
+    current_reporter = reset_reporter(linter, output)
 
     # Try to check file, issue error message for invalid files.
     try:
-        for locations in _get_valid_files_to_check(module_name, local_config):
-            for file_py in _get_file_paths(locations):
+        for locations in _get_valid_files_to_check(current_reporter, module_name, local_config):
+            for file_py in current_reporter.get_file_paths(locations):
                 # Load config file in user location. Construct new linter each
                 # time, so config options don't bleed to unintended files.
-                linter = _init_linter(config=local_config, file_linted=file_py)
+                linter = reset_linter(config=local_config, file_linted=file_py)
                 # The local config may have set a new reporter
-                current_reporter = _init_reporter(linter)
-                current_reporter.show_file_linted(file_py)
+                current_reporter = reset_reporter(linter)
```

Comment:
I was originally going to say that this function call seems to be missing an argument. But actually, let's say that unlike linters, a new reporter is *not* created for each file, so calling `reset_reporter` should be unnecessary (just register it with the new linter).